Tasks/AzureTestPlanV0/testLibExecutor.ts

```diff
@@ -149,12 +162,16 @@ export async function execGradleBuild(args: string[]): Promise<number> {
     gradleRunner.arg('clean');
     gradleRunner.arg(args);
 
+    let runnerOutput = { stdout: ''};
+
     try {
-        await gradleRunner.exec(getExecOptions());
+        await gradleRunner.exec(getExecOptions(runnerOutput));
         // Gradle build succeeded
         return 0; // Return 0 indicating success
     } catch (err) {
-        console.error(err.message);
+        if (runnerOutput.stdout.includes('There were failing tests')) {
+            return 0;
+        }
         tl.setResult(tl.TaskResult.Failed, "Build failed."); // Set the step result to Failed
```

Comment:
Unrelated to changes, but for all the warning and error message, we have to create a message string in `task.json` file to support localization. Please create a work item for it and lets prioritize in the next sprint.

Reply:
Sure, I created work item and will work on it in the upcoming sprint, https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/2236462